cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/TopicReplicaDistributionGoal.java

```diff
@@ -246,11 +248,15 @@ protected void initGoalState(ClusterModel clusterModel, OptimizationOptions opti
     Set<String> topicsToRebalance = topicsToRebalance(clusterModel, optimizationOptions.excludedTopics());
 
     // Initialize the average replicas on an alive broker.
-    for (String topic : topicsToRebalance) {
+    for (String topic : clusterModel.topics()) {
       int numTopicReplicas = clusterModel.numTopicReplicas(topic);
       _avgTopicReplicasOnAliveBroker.put(topic, (numTopicReplicas / (double) clusterModel.aliveBrokers().size()));
       _balanceUpperLimitByTopic.put(topic, balanceUpperLimit(topic, optimizationOptions));
       _balanceLowerLimitByTopic.put(topic, balanceLowerLimit(topic, optimizationOptions));
+      // Retain only the topics to rebalance in _avgTopicReplicasOnAliveBroker
+      if (!topicsToRebalance.contains(topic)) {
```

Comment:
Instead of add then remove topic from `_avgTopicReplicasOnAliveBroker`, maybe we can move the if at line 253 to not add in the first place?

Reply:
We add it initially to ensure that `balanceUpperLimit` and `balanceLowerLimit` can use the corresponding average values.

Reply:
gotcha.